### DISCUSSION TITLE
Writing to Solis Inverter

### DISCUSSION THREAD

**Discussion by matchermike1981** [wills106/homeassistant-solax-modbus#484] [Q&A] [answered]
Hey all,

I'm running this integration and it pulls my solis inverter stats via modbus great. I'm unclear how i write back to the unit? I'd like to change times of charging etc. Is there a really simple guide on how to achieve this please?

Thanks
Mike

**Comment by wills106** [comment, marked as answer]
There is a description in the [Wiki](https://github.com/wills106/homeassistant-solax-modbus/wiki/Ginlong-Solis-Modes-of-Operation)

**Reply by matchermike1981** [reply]
Sorry. I knew it would be a silly question. I was missing the controls panel and thus unable to work out how to change values. Thank you for such a great integration and quick response.

**Reply by matchermike1981** [reply]
Is it possible to change out of "Timed Charge/Discharge - No Grid Charging" mode? The value for this reverts even after pressing Sync or update buttons.

The times change as required. I want to run self use, so charging from solar throughout the day, but modify grid charge times to be either 0 or a chosen period. 

Thanks

**Reply by wills106** [reply]
Does the Select entity "Energy Storage Control Switch" not work for you in changing the Modes of operation?
There should be 8 options to select from.

Or is it the select that reverts back to the default?

**Reply by matchermike1981** [reply]
I can see these 8 options, and change between "Selfuse - no grid charging" and "Timed Charge/Discharge - No grid charging". This seems to change the mode correctly (i can see this as the value changes in HA for the read from inverter value).

"Selfuse" and "Timed Charge/Discharge" change after 5/10 seconds to the "No Grid charging" variants automatically.

Offgrid mode puts the value into "Unknown" (but i don't need that).
